pkg/frontend/middleware/audit.go

```diff
@@ -0,0 +1,134 @@
+package middleware
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/log"
+	"github.com/Azure/ARO-RP/pkg/util/log/audit"
+)
+
+// Audit generates an audit log based on the request, caller, resource and
+// correlation data found in a HTTP request. It depends on the 'Log' middleware
+// to populate the request context with data that it needs.
+func Audit(env env.Core, entry *logrus.Entry) func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// skip requests to the 'operationsstatus' endpoints
+			if strings.Contains(r.URL.Path, "operationsstatus") {
```

Comment:
If someone names their cluster "operationsstatus"?
This check feels somewhat arbitrary - I wonder whether we would be better off dropping it, or maybe deciding not to audit log all GET (r/o) requests?
We will already be writing 2 loglines already for every operationsstatus call, so the fact that there is a 3rd audit log doesn't bother me massively in terms of log explosion.

Reply:
it is quite arbitrary, honestly. if we don't mind the noise i don't mind removing this check. i think there are at least 6 of these `operationstatus` calls during `az aro create`.

as for removing all `GET` requests, will `GET` secret requests ever made it to the frontend, or are they blocked somewhere else? do we want to log those?